packages/auth/src/signature/middleware.ts

```diff
@@ -151,6 +154,15 @@ export async function tokenHttpsigMiddleware(
   const grantService = await ctx.container.use('grantService')
   const grant = await grantService.get(accessToken.grantId)
 
+  if (!grant) {
+    ctx.status = 401
+    ctx.body = {
+      error: 'invalid_interaction',
+      message: 'invalid grant'
+    }
```

Comment:
test me

Reply:
Same question as here: https://github.com/interledger/rafiki/pull/921#discussion_r1067951869

Reply:
In this case maybe the query in `getByManagementId` can include `.withGraphFetched('grant')`

Reply:
Also did that but same question as above.

Reply:
`accessToken.grant` should always be defined.
You could log and return `500` (which you could test by mocking `accessTokenService.getByManagementId`) or change the `accessTokenService.getByManagementId` return type to an `AccessTokenWithGrant`